docs/development/synapse_architecture/streams.md

```diff
@@ -0,0 +1,126 @@
+## Streams
+
+Synapse has a concept of "streams", which are roughly described in [`id_generators.py`](
+    https://github.com/matrix-org/synapse/blob/develop/synapse/storage/util/id_generators.py
+).
+It is important to understand this, so let's describe them formally.
+We paraphrase from the docstring of [`AbstractStreamIdGenerator`](
+    https://github.com/matrix-org/synapse/blob/a719b703d9bd0dade2565ddcad0e2f3a7a9d4c37/synapse/storage/util/id_generators.py#L96
+).
+
+A stream is an append-only log `T1, T2, ..., Tn, ...` of facts which grows over time.
+Only "writers" can add facts to a stream, and there may be multiple writers.
```

Comment:
That's right, `event` is something to be avoided.

I originally called these "transactions" based on (your?) docstrings in the AbstractStreamIDGenerator. I think Erik suggested the term fact. I think it works well because once a fact is added to a stream and completes, it is never altered posthumously.